pkgs/development/python-modules/commitizen/default.nix

```
@@ -9,7 +9,7 @@
 , testers
 }:
 
-python3.pkgs.buildPythonApplication rec {
+python3.pkgs.buildPythonPackage rec {
```

Should be

```suggestion
buildPythonPackage rec {
```

for Python library.
You need to import buildPythonPackage and all the Python dependencies.

---

Same for nativeBuildInputs, propagatedBuildInputs, nativeCheckInputs.

Example:

```
nativeBuildInputs = with python3.pkgs; [
  poetry-core
  installShellFiles
];
```

should become for Python library:

```nix
{
  argcomplete,
  charset-normalizer,
  colorama,
  decli,
  importlib-metadata,
  jinja2,
  packaging,
  pyyaml,
  questionary,
  termcolor,
  tomlkit,
}:

propagatedBuildInputs = with python3.pkgs; [
  argcomplete
  charset-normalizer
  colorama
  decli
  importlib-metadata
  jinja2
  packaging
  pyyaml
  questionary
  termcolor
  tomlkit
];
```

---

Done in https://github.com/NixOS/nixpkgs/pull/329638/commits/3b1b6a1d40413af633271c63b18af26152623652